src/quickgui/qgsquickmapsettings.cpp

```diff
@@ -118,7 +118,10 @@ QPointF QgsQuickMapSettings::coordinateToScreen( const QgsPoint &point ) const
 
 QgsPoint QgsQuickMapSettings::screenToCoordinate( const QPointF &point ) const
 {
-  const QgsPointXY pp = mMapSettings.mapToPixel().toMapCoordinates( point.toPoint() );
+  // use floating point precision with mapToCoordinatesF
+  // this is to aoviding rounding errors with an odd screen width or height
+  // and the point being set to the exact center of it
+  const QgsPointXY pp = mMapSettings.mapToPixel().toMapCoordinatesF( point.x(), point.y() );
```

Comment:
Why not rename `QgsMapToPixel().toMapCoordinatesF(double, double);` to `QgsMapToPixel().toMapCoordinates(double, double);`?
This should overload nicely and avoid this class of errors.
It can still be tagged with a PyName `toMapCoordinatesF`, but even there it might be nice to always call the float version (and keep `toMapCoordinatesF` as an alias until QGIS 4.0)

Reply:
@m-kuhn if we give float values, it will automatically prefer the double precision signature over the integer one?

Reply:
Yes it will.
https://stackoverflow.com/questions/46150905/promotion-of-parameters-during-overloading

> To approximate our notions of what is reasonable, a series of criteria are tried in order: 1 Exact match **[2] Match using promotions**;  [3] Match using standard conversions [4] Match using user-defined conversions [5] Match using the ellipsis ......

and

> [conv.fpprom]: (7.7 Floating-point promotion )
>
>    A prvalue of type float can be converted to a prvalue of type double. The value is unchanged.
>    This conversion is called floating-point promotion.